azurerm/internal/services/dns/dns_zone_data_source.go

```diff
@@ -109,26 +107,33 @@ func dataSourceArmDnsZoneRead(d *schema.ResourceData, meta interface{}) error {
 	return tags.FlattenAndSet(d, resp.Tags)
 }
 
-func findZone(client *dns.ZonesClient, rgClient *resources.GroupsClient, ctx context.Context, name string) (dns.Zone, string, error) {
-	groups, err := rgClient.List(ctx, "", nil)
+func findZone(client *dns.ZonesClient, ctx context.Context, name string) (dns.Zone, string, error) {
```

Comment:
thinking about it this probably should be a pointer:

```suggestion
func findZone(client *dns.ZonesClient, ctx context.Context, name string) (*dns.Zone, string, error) {
```